lib/flight_job/models/job.rb

```diff
@@ -173,7 +178,7 @@ def state
       when 'FAILED_SUBMISSION'
         'FAILED'
       else
-        metadata.state || 'UNKNOWN'
+        valid? ? metadata.state || 'UNKNOWN' : broken_metadata.state
```

Comment:
Did you come across any issues requiring this line to be added?  If we remove the call to `broken_metadata` from `Job#state` then the only class using `BrokenMetadata` is `JobDecorator`.  `BrokenMetadata` at that point is used only during the display of jobs.

We could also replace `UNKNOWN` with `BROKEN`.  The job's state being unknown sounds quite broken to me.

Reply:
I changed this line to keep the behaviour of the code the same as much as possible, except for introducing the broken state for invalid jobs. So there wasn't a particular situation this was built around - I've changed it